File: JAVA/Irrgarten_JAVA/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alejandro
 */
public class Weapon {
    private float power;
        private int uses;
        
        public Weapon(float power, int uses){
            this.power=power;
            this.uses=uses;
        }
        
        public float attack(){
            if(uses>0){
                uses--;
                return power;
            }else{
                return 0;
            }
        }
        
        public String toString(){
            return "W[" + power + "," + uses + "]";
        }
        
        public float getPower(){
            return power;
        }
        
        public int getUses(){
            return uses;
        }
        
        public boolean discard(){
            return Dice.discardElement(uses);
        }
}

File: RUBY/PRACTICA1/Practica-1/irrgarten_1/weapon.rb
# frozen_string_literal: true
module Irrgarten
  class Weapon
    attr_reader :power, :uses

    def initialize(power,uses)
      @power = power.to_f
      @uses = uses.to_i
    end

    def attack
      if @uses > 0
        @uses -= 1
        @power
      else
        0
      end
    end

    def to_s
      "W[#{@power},#{@uses}]"
    end

    def discard
      Dice.discard_element(@uses)
    end
  end
end